impeller/tessellator/tessellator.cc

```diff
@@ -243,29 +242,38 @@ Tessellator::TessellateConvex(const Path& path, Scalar tolerance) {
                             point_buffer_ = std::move(point_buffer);
                           });
 
+  output.reserve(polyline.points->size() +
+                 (4 * (polyline.contours.size() - 1)));
   for (auto j = 0u; j < polyline.contours.size(); j++) {
     auto [start, end] = polyline.GetContourPointBounds(j);
-    auto center = polyline.GetPoint(start);
+    auto origin = polyline.GetPoint(start);
 
     // Some polygons will not self close and an additional triangle
     // must be inserted, others will self close and we need to avoid
     // inserting an extra triangle.
-    if (polyline.GetPoint(end - 1) == polyline.GetPoint(start)) {
+    if (polyline.GetPoint(end - 1) == origin) {
       end--;
     }
-    output.emplace_back(center);
-    output.emplace_back(polyline.GetPoint(start + 1));
-
-    for (auto i = start + 2; i < end; i++) {
-      const auto& point_b = polyline.GetPoint(i);
-      output.emplace_back(point_b);
+    if (j > 0) {
+      // Triangle strip break.
+      output.emplace_back(output.back());
+      output.emplace_back(origin);
+      output.emplace_back(origin);
+      output.emplace_back(polyline.GetPoint(start + 1));
+    } else {
+      output.emplace_back(origin);
+    }
 
-      indices.emplace_back(0);
-      indices.emplace_back(i - 1);
-      indices.emplace_back(i);
+    size_t a, b;
+    for (a = start + 1, b = end - 1; a < b; a++, b--) {
+      output.emplace_back(polyline.GetPoint(a));
+      output.emplace_back(polyline.GetPoint(b));
+    }
+    if (a == b) {
+      output.emplace_back(polyline.GetPoint(a));
```

Comment:
I think this checks out... for 3 points it should append indices 0, 1, 2, and for 4 points it should append 0, 1, 3, 2, etc.

Reply:
Ahh but there is a bug here, on L268 b should start at `end` and not `end -`, probably ...

Reply:
or maybe not, let me double check my assumptions here

Reply:
YEah, there was an issue in how I was doing the strip breaks, I needed to remove an extra point, now the triangles show up. Good idea, once again